`ember help generate` fails when ember-intl is installed

When using `ember help generate`, ember-intl fails to report its generators with the following error.

```
Cannot find module '../../lib/locale-writer'
Error: Cannot find module '../../lib/locale-writer'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/sean/source/influencer-search/node_modules/ember-intl/blueprints/translation/index.js:6:20)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

--- a/app/initializers/ember-intl.js
+++ b/app/initializers/ember-intl.js
@@ -3,7 +3,7 @@
  * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
  */
 
-export function initializer(app) {
+export function initializer() {
   // intentionally blank
 }
 

--- a/blueprints/translation/index.js
+++ b/blueprints/translation/index.js
@@ -3,17 +3,16 @@
  * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
  */
 
-var broccoliCLDR = require('../../lib/locale-writer');
+var isSupportedLocale = require('../../lib/utils/is-supported-locale');
 var Blueprint = require('ember-cli/lib/models/blueprint');
-var SilentError = require('ember-cli/lib/errors/silent');
+var SilentError = require('silent-error');
 
 module.exports = {
   description: 'Adds an empty translation file and locale is supported',
 
   normalizeEntityName: function (localeName) {
     entityName = Blueprint.prototype.normalizeEntityName.apply(this, arguments);
-
-    if (!broccoliCLDR.has(localeName)) {
+    if (!isSupportedLocale(localeName)) {
       throw new SilentError('Aborting. `' + localeName + '` is not a know locale');
     }
 

--- a/index.js
+++ b/index.js
@@ -163,9 +163,7 @@ module.exports = {
     if (existsSync(translations)) {
       locales = walkSync(translations).map(function(filename) {
         return path.basename(filename, path.extname(filename));
-      }).filter(function(locale) {
-        return CldrWriter.has(locale);
-      });
+      }).filter(utils.isSupportedLocale);
     }
 
     if (this.addonOptions.locales) {

--- a/lib/broccoli/cldr-writer.js
+++ b/lib/broccoli/cldr-writer.js
@@ -12,6 +12,7 @@ var cldr = require('formatjs-extract-cldr-data');
 var mkdirp = require('mkdirp');
 var path = require('path');
 var fs = require('fs');
+
 var assign = require('../utils/assign');
 
 var CldrWriter = CachingWriter.extend({
@@ -51,15 +52,5 @@ var CldrWriter = CachingWriter.extend({
     });
   }
 });
-
-CldrWriter.has = function has() {
-  try {
-    return !!cldr({
-      locales: [locale]
-    });
-  } catch (err) {
-    return false;
-  }
-}
 
 module.exports = CldrWriter;

--- a/lib/utils/index.js
+++ b/lib/utils/index.js
@@ -7,5 +7,6 @@
   makeArray: require('./make-array'),
   uniqueByString: require('./unique-by-string'),
   isModern: require('./is-modern'),
-  assign: require('./assign')
+  assign: require('./assign'),
+  isSupportedLocale: require('./is-supported-locale')
 }

--- a/lib/utils/is-supported-locale.js
+++ b/lib/utils/is-supported-locale.js
@@ -1,0 +1,12 @@
+var fs = require('fs');
+var path = require('path');
+
+var dataPath = path.join(require.resolve('formatjs-extract-cldr-data'), '..', 'data', 'main');
+
+var availableLocales = fs.readdirSync(dataPath).map(function(localeFile) {
+  return localeFile.toLowerCase();
+});
+
+module.exports = function(locale) {
+  return availableLocales.indexOf(locale.toLowerCase()) > -1
+}

--- a/package.json
+++ b/package.json
@@ -43,6 +43,7 @@
     "mkdirp": "^0.5.0",
     "object.assign": "^4.0.1",
     "serialize-javascript": "^1.0.0",
+    "silent-error": "^1.0.0",
     "walk-sync": "^0.1.3"
   },
   "devDependencies": {

